File: Data Structures & Algorithms/is-palindrome/submission-1.js
class Solution {
    isPalindrome(s) {
        let left = 0;
        let right = s.length - 1;

        while (left < right) {
            // skip non-alphanumeric
            while (left < right && !/[a-z0-9]/i.test(s[left])) left++;
            while (left < right && !/[a-z0-9]/i.test(s[right])) right--;

            if (s[left].toLowerCase() !== s[right].toLowerCase()) {
                return false;
            }

            left++;
            right--;
        }

        return true;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] left=new int[nums.length];
        int product = 1;
        for(int i=nums.length-1;i>=0;i--){
            left[i]=product;
            product=product*nums[i];
        }
        int right[]=new int[nums.length];
        product=1;
        for(int i=0;i<nums.length;i++){
            right[i]=product;
            product=product*nums[i];
        }
        int[] result=new int[nums.length];
        for(int i=0;i<nums.length;i++){
            result[i]=left[i]*right[i];
        }
        return result;
    }
}  
